### DISCUSSION TITLE
graphic elements X coordinate

### DISCUSSION THREAD

**Discussion by jppeltier** [Mathieu2301/TradingView-API#38] [Q&A] [answered]
Hello Mathieu,

So, I have installed V3, and I confirm session handling is working flawlessly so far, well done! I however have something else troubling me regarding `graphic` elements. I don't expect it to be a bug from your scripts, but this issue might significantly impair the true potential of your tool.

Let's use the BINANCE:BTCUSDT daily chart, with Zig-Zag in-house indicator (code: `"STD;Zig_Zag"`). I found, in the study `graphic` element labels and lines corresponding to the Zig-Zag indicator as displayed. So, for such specific chart, we find indeed the following first elements:
- a label (in `graphic.labels`):
```
{
      id: 2,
      x: 0,
      y: 3400,
      yl: 'bl',
      t: '3400.00 (+3400.00) \n4.502K',
      st: 'n',
      ci: 1,
      tci: 2,
      sz: 'normal',
      ta: 'center',
      tt: ''
    }
```
; and
- a following line (in `graphic.lines`):
```
{
      id: 3,
      x1: 0,
      y1: 3400,
      x2: 1,
      y2: 4939.19,
      ex: 'n',
      st: 'sol',
      ci: 0,
      w: 2
}
```
If you check the TV interface, you will indeed find such, first label and line at the very beginning of quotations, in August-September 2017. However, the reported `x`, `x1`, `x2` seem to be only an increasing series of coordinates (0, 1, 2, 3 ...) not matching either the dates, nor even the number of bars! By comparing in the TV interface, we should have:
- `x: 0` is the 6th bar, corresponding to 22 August 2017 ; and
- `x: 1` is the 17th bar, corresponding to 2 September 2017.

![BTCUSDT_2021-12-23_16-13-17](https://user-images.githubusercontent.com/58844169/147245719-01526477-3a05-4b68-afe4-d03416b91d84.jpeg)

Hence my question: how can we match back `x` ordinal value with the number of bars or, better, the actual timestamp?

_(tip: I checked `graphic.indexes` hoping to find my answer, but it remains desperately `undefined` ...)_

Thanks!

**Comment by Mathieu2301** [comment, marked as answer]
Don't worry 😉
We are working on it in #17, I just have to translate all the abbreviations like 't', 'sz', 'ta', etc... to 'title', 'size', 'textAlign', etc... And do all the JSDoc. (You can help translate if you want)
